msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/stress/AbstractMsalUiStressTest.java

```diff
@@ -0,0 +1,256 @@
+    }
+
+    private synchronized void updateTestPassRate(boolean passed) {
+        if (passed) {
+            testsPassed++;
+        } else {
+            testsFailed++;
+        }
+    }
+
+    private Runnable getPerformanceStatsCollector() {
+        return new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    while (true) {
+                        // time   cpu_usage   memory_usage   data_received   data_sent   num_threads   time_limit   tests_passed   tests_failed   device_memory   device_name
+                        String line = String.format(
+                                "%s,%.2f,%d,%d,%d,%d,%d,%d,%d,%d,%s",
+                                new SimpleDateFormat(DATE_FORMAT).format(new Date()),
+                                DeviceMonitor.getCpuUsage(),
+                                DeviceMonitor.getMemoryUsage(),
+                                DeviceMonitor.getNetworkTrafficInfo().getDiffBytesReceived(),
+                                DeviceMonitor.getNetworkTrafficInfo().getDiffBytesSent(),
+                                getNumberOfThreads(),
+                                getTimeLimit(),
+                                testsPassed,
+                                testsFailed,
+                                DeviceMonitor.getTotalMemory(),
+                                DeviceMonitor.getDeviceName()
+                        );
+
+                        writeFile(line);
+
+                        TimeUnit.MILLISECONDS.sleep(DEVICE_MONITOR_INTERVAL);
+                    }
+                } catch (Exception ex) {
+                    executionException = ex;
+                }
+            }
+        };
+    }
+
+    private synchronized void writeFile(final String output) {
+        try {
+            final FileAppender fileAppender = new FileAppender(getOutputFileName(), new LogcatLikeFormatter());
+            fileAppender.append(output);
+
+            CommonUtils.copyFileToFolderInSdCard(fileAppender.getLogFile(), "automation");
+        } catch (IOException e) {
+            executionException = e;
+        }
+    }
+
+
+    @Override
+    public String[] getScopes() {
+        return new String[]{"User.read"};
+    }
+
+    @Override
+    public String getAuthority() {
+        return mApplication.getConfiguration().getDefaultAuthority().toString();
+    }
+
+    @Override
+    public int getConfigFileResourceId() {
+        return R.raw.msal_config_webview;
+    }
+
+    @Override
+    public LabUserQuery getLabUserQuery() {
+        final LabUserQuery query = new LabUserQuery();
+        query.azureEnvironment = LabConstants.AzureEnvironment.AZURE_CLOUD;
+
+        return query;
+    }
+
+    @Override
+    public String getTempUserType() {
+        return null;
+    }
+
+    @Override
+    public RuleChain getPrimaryRules() {
+        // Use the time limit as the test timeout rule and add 5 more minutes for cleaning up of the tests
```

Comment:
Why was 5 minutes decided?

Reply:
It is a just a placeholder at the moment, I'm open to suggestions.